ecommerce/serializers.py

```diff
@@ -109,10 +94,39 @@ class Meta:
             "thumbnail_url",
             "object_id",
             "product_id",
+            "readable_id",
         ]
         model = models.ProductVersion
 
 
+class ProductSerializer(serializers.ModelSerializer):
+    """ Product Serializer """
+
+    title = serializers.SerializerMethodField()
+    product_type = serializers.SerializerMethodField()
+    latest_version = serializers.SerializerMethodField()
+
+    def get_title(self, instance):
+        """ Return the product title """
+        return instance.content_type.get_object_for_this_type(
+            pk=instance.object_id
+        ).title
+
+    def get_product_type(self, instance):
+        """ Return the product type """
+        return instance.content_type.model
+
+    def get_latest_version(self, instance):
+        """Serialize and return the latest ProductVersion for the Product"""
+        return ProductVersionSerializer(
+            instance.latest_version, context={**self.context, "show_all_runs": True}
+        ).data
```

Comment:
This serializer is used by other views etc. Would it be worth it to create a separate serialize (`ProductDetialSerializer` et al) so that the other clients of this serializer don't need to run this extra query to fetch the latest product version?